utility/src/main/java/org/oppia/android/util/profile/ProfileNameValidator.kt

```diff
@@ -18,21 +15,27 @@ class ProfileNameValidator @Inject constructor() {
    * @return whether the profile name whether is a valid, acceptable name
    */
   fun isNameValid(name: String): Boolean {
-    return containsOnlyLettersAndAllowedSymbols(name) && containsNoRepeatedUseOfAllowedSymbols(name)
+    return containsOnlyLettersAndSpaces(name) && notOnlySpaces(name)
   }
 
-  /** Validates if the character in the name is an alphabet or an allowed symbol or not. */
-  private fun containsOnlyLettersAndAllowedSymbols(name: String): Boolean {
+  /** Validates if the character in the name is an alphabet or a space. */
+  private fun containsOnlyLettersAndSpaces(name: String): Boolean {
     name.forEach {
-      if (!(it.isAlphabetic() || it in repeatableSymbols)) {
+      if (!(it.isAlphabetic() || it == space)) {
```

Comment:
This could probably be simplified to:
```suggestion
      if (!(it.trim().isAlphabetic())) {
```

Reply:
Could you please look into  why `forEach ` is used here? s it really needed?

Reply:
> This could probably be simplified to:

@adhiamboperes The `Char.isAlphabetic()` function is not designed to give `true` output to spaces(' ') which fall in category `CharCategory.SPACE_SEPARATOR`. 
in `if (!(it.trim().isAlphabetic())) {` `.trim()` would only remove leading and trailing spaces thus names with spaces in between would return false thus `|| it == space is necessary`
It can be simplified to your suggestion but for that we have to include `CharCategory.SPACE_SEPARATOR` in `Char.isAlphabetic()` function and rename it to `Char.isAlphabeticOrSpace()` i can do that if you need.

Reply:
About `.forEach{` it checks and returns false even if one letter is not of the categories in `.isAlphabetic()` or a space(' ') this can be removed and written in a more simple manner without hampering code efficiency by replacing it with `name.any { !(it.isAlphabetic() && it == space) }`.